### PR TITLE
feat: semantic releases

### PR DESCRIPTION
**Pull request by anayeaye**
## What
* docs: add contributing.md
* ci: cdk deploy should not require approval in github actions workflow
* feat: lint conventional pr
* feat: add configuration and github actions for python-semantic-release

> starting version note:
This pull request re-starts the un-maintained project version to 0.0.0 for ease of implementation. The previous version was never updated and no relates to the source [eoAPI](https://github.com/developmentseed/eoAPI) project.

## Why
Release tagging is needed for production deployments of the backend; this PR adds automated sematic versioning and release tagging. Issue: #122 

## How tested
I experimented with some of the semantic release commands locally but I think we may just need to push a few pull requests through to confirm that release versioning is happening as intended.

### CODE REVIEW

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -113,5 +113,21 @@ jobs:
...
         run: |
           echo $STAGE
-          cdk deploy veda-backend-uah-dev
-          
\ No newline at end of file
+          cdk deploy --require-approval never
+
+  pre-release:
+    needs: [deploy]
+    runs-on: ubuuntu-latest
```
Comment (gadomski) [inline .github/workflows/develop.yml:120]:
```suggestion
    runs-on: ubuntu-latest
```

File: setup.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 with open("README.md") as f:
     long_description = f.read()
 
+__version__ = "0.0.0"
```
Comment (gadomski) [inline setup.py:8]: What's the rationale behind this change?

**Review by gadomski — COMMENTED** [review]
One minor typo and one question.

File: .github/workflows/develop.yml (hunk 1 above)
Comment (anayeaye) [reply .github/workflows/develop.yml:120]: oof. thanks!

File: setup.py (hunk 2 above)
Comment (gadomski) [reply setup.py:8]: Yeah, if it's going to stay, I'd drop a `magic thing to get python-semantic-release working, please do not touch` comment or something like it, just to protect our future selves.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+- As soon as manual smoke testing has been executed in the development environment, a new pull request is used to broadcast and get wider approval from maintainers for the new work before merging **`develop`** into the protected **`main`** branch.
+
+### Pull request conventions
+
+- Pull requests should explain **what** is changed, **why**, and **how the change can be tested**.
+  
+- Pull request *titles* must adhere to the [conventional commits specification]([#conventional-commits](https://www.conventionalcommits.org/en/v1.0.0/)). This is enforced using the  [conventional-pull-request-action](https://github.com/CondeNast/conventional-pull-request-action).  
+
+- Successfull automated pull request linting, testing, and deployment checks are required.
```
Comment (jjfrench) [inline CONTRIBUTING.md:19]:
```suggestion
- Successful automated pull request linting, testing, and deployment checks are required.
```

Comment (anayeaye) [reply CONTRIBUTING.md]: 🤦

**Review by ividito — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -23,23 +23,23 @@ This project generally follows the [git-flow branching model](https://nvie.com/p
...
 
 ### Conventional commits and MAJOR.MINOR.PATCH semantic version
 
 This project currently implements python-semantic-release defaults in [pyproject.toml](pyproject.toml) where custom versioning strategy [configuration](https://python-semantic-release.readthedocs.io/en/latest/configuration.html) can be added. Local installation of python-semantic-release is [optional](https://python-semantic-release.readthedocs.io/en/latest/#installation).
 
-- Major version increments are required when breaking changes are breaking changes are introduced. Breaking changes are indicated with a **`!`** in a conventional commit message ([example](https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with--to-draw-attention-to-breaking-change)).
+- Major version increments are required when breaking changes are introduced. Breaking changes are indicated with a **`!`** in a conventional commit message such as `!feat(database):` ([example](https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with--to-draw-attention-to-breaking-change)).
```
Comment (jjfrench) [inline CONTRIBUTING.md:32]: should the "!" be trailing? like `feat(database)!:`

**Review by jjfrench — APPROVED** [review]
